src/core/ipe-setup.c

```diff
@@ -0,0 +1,127 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "copy.h"
+#include "fd-util.h"
+#include "fileio.h"
+#include "ipe-setup.h"
+
+#define IPE_SECFS_DIR "/sys/kernel/security/ipe"
+#define IPE_SECFS_NEW_POLICY IPE_SECFS_DIR "/new_policy"
+#define IPE_SECFS_POLICIES IPE_SECFS_DIR "/policies/"
+#define IPE_POLICY_PATH "/etc/ipe/ipe-policy.p7b"
+
+int ipe_setup(void) {
+#if ENABLE_IPE
+        _cleanup_free_ char *policy_name = NULL, *policy_version = NULL, *policy_existing_version = NULL,
+                *policy_existing_version_file = NULL, *policy_activate = NULL, *output_path_free = NULL;
+        _cleanup_fclose_ FILE *finput = NULL;
+        _cleanup_close_ int input = -EBADF, output = -EBADF;
+        const char *output_path;
+        int r;
+
+        if (access(IPE_SECFS_DIR, F_OK) < 0) {
+                log_debug_errno(errno, "IPE support is disabled in the kernel, ignoring: %m");
+                return 0;
+        }
+
+        if (access(IPE_POLICY_PATH, F_OK) < 0) {
+                log_debug_errno(errno, "No IPE custom policy file "IPE_POLICY_PATH", ignoring: %m");
+                return 0;
+        }
+
+        input = open(IPE_POLICY_PATH, O_RDONLY|O_CLOEXEC);
+        if (input < 0)
+                return log_error_errno(errno, "Failed to open the IPE custom policy file "IPE_POLICY_PATH", ignoring: %m");
+
+        r = fdopen_independent(input, "re", &finput);
+        if (r < 0)
+                return log_error_errno(r, "Failed to fdopen the IPE custom policy file "IPE_POLICY_PATH", ignoring: %m");
+
+        /* First we need to parse the policy name, since in order to activate it we need to know it, as it
+         * will show up as a directory under /sys/kernel/security/ipe/policies/ and there will be an
+         * 'active' file under it to write into, which will make it active. */
+        for (;;) {
+                _cleanup_free_ char *line = NULL;
+                char *p;
+
+                r = read_line(finput, LONG_LINE_MAX, &line);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to read the IPE policy "IPE_POLICY_PATH": %m");
+                if (r == 0)
+                        break;
+
+                p = strstrafter(line, "policy_name=");
+                if (p && !policy_name) {
+                        policy_name = strdup(p);
+                        if (!policy_name)
+                                return log_oom();
+                        p = strchr(policy_name, ' ');
+                        if (p)
+                                *p = '\0';
```

Comment:
hmm, strchr? not strrchr?

delete_trailing_chars()

Reply:
> hmm, strchr? not strrchr?

Yes? A single line can have multiple space-separated key=value pairs, so we want the first, not the last

Reply:
> delete_trailing_chars()

I don't think so? I want to actually truncate at the first space, if any. See: https://microsoft.github.io/ipe/usage/examples/

```
policy_name=Allow_All_Initramfs policy_version=0.0.0
DEFAULT action=DENY

op=EXECUTE boot_verified=TRUE action=ALLOW
```

So when encountering policy_name=, I want to truncate at the space before policy_version